src/vs/workbench/contrib/terminal/browser/terminalProfileService.ts

```diff
@@ -0,0 +1,225 @@
+	private readonly _primaryOffProcessTerminalService?: IOffProcessTerminalService;
+
+	private readonly _onDidChangeAvailableProfiles = new Emitter<ITerminalProfile[]>();
+	get onDidChangeAvailableProfiles(): Event<ITerminalProfile[]> { return this._onDidChangeAvailableProfiles.event; }
+
+	get profilesReady(): Promise<void> { return this._profilesReadyBarrier.wait().then(() => { }); }
+	get availableProfiles(): ITerminalProfile[] {
+		this.refreshAvailableProfiles();
+		return this._availableProfiles || [];
+	}
+	get contributedProfiles(): IExtensionTerminalProfile[] {
+		return this._contributedProfiles || [];
+	}
+	constructor(
+		@IContextKeyService private readonly _contextKeyService: IContextKeyService,
+		@IConfigurationService private readonly _configurationService: IConfigurationService,
+		@ITerminalContributionService private readonly _terminalContributionService: ITerminalContributionService,
+		@IExtensionService private readonly _extensionService: IExtensionService,
+		@IRemoteAgentService private _remoteAgentService: IRemoteAgentService,
+		@IWorkbenchEnvironmentService private readonly _environmentService: IWorkbenchEnvironmentService,
+		@IRemoteTerminalService private readonly _remoteTerminalService: IRemoteTerminalService,
+		@optional(ILocalTerminalService) private readonly _localTerminalService: ILocalTerminalService
+	) {
+		// in web, we don't want to show the dropdown unless there's a web extension
+		// that contributes a profile
+		this._extensionService.onDidChangeExtensions(() => this.refreshAvailableProfiles());
+
+		this._configurationService.onDidChangeConfiguration(async e => {
+			const platformKey = await this._getPlatformKey();
+			if (e.affectsConfiguration(TerminalSettingPrefix.DefaultProfile + platformKey) ||
+				e.affectsConfiguration(TerminalSettingPrefix.Profiles + platformKey) ||
+				e.affectsConfiguration(TerminalSettingId.UseWslProfiles)) {
+				await this.refreshAvailableProfiles();
+			}
+		});
+		this._webExtensionContributedProfileContextKey = TerminalContextKeys.webExtensionContributedProfile.bindTo(this._contextKeyService);
+
+		this._primaryOffProcessTerminalService = !!this._environmentService.remoteAuthority ? this._remoteTerminalService : (this._localTerminalService || this._remoteTerminalService);
+		// Wait up to 5 seconds for profiles to be ready so it's assured that we know the actual
+		// default terminal before launching the first terminal. This isn't expected to ever take
+		// this long.
+		this._profilesReadyBarrier = new AutoOpenBarrier(5000);
+		this.refreshAvailableProfiles();
+	}
+
+	_serviceBrand: undefined;
+
+	getDefaultProfileName(): string {
+		if (!this._defaultProfileName) {
+			throw new Error('no default profile');
+		}
+		return this._defaultProfileName;
+	}
+
+	@throttle(2000)
+	refreshAvailableProfiles(): void {
+		this.refreshAvailableProfilesNow();
+	}
+
+	async refreshAvailableProfilesNow(): Promise<void> {
```

Comment:
@Tyriar  had to expose this because without awaiting this, the tests wouldn't work

Reply:
make it protected and make a new function that calls and awaits it